app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -172,19 +175,35 @@ private class HomeTileAdapter(
 private fun onBindBundledHomeTile(holder: TileViewHolder, tile: BundledHomeTile) = with (holder) {
     val bitmap = BundledTilesManager.getInstance(itemView.context).loadImageFromPath(itemView.context, tile.imagePath)
     iconView.setImageBitmap(bitmap)
+
+    titleView.text = tile.title
 }
 
 private fun onBindCustomHomeTile(uiLifecycleCancelJob: Job, holder: TileViewHolder, item: CustomHomeTile) = with (holder) {
-    launch(uiLifecycleCancelJob + CommonPool) {
-        val screenshot = HomeTileScreenshotStore.read(itemView.context, item.id) // TODO: if null, provide placeholder.
-        launch(UI) {
-            // Animate to avoid pop-in due to thread hand-offs. TODO: animation is janky.
-            ObjectAnimator.ofInt(iconView, "imageAlpha", 0, 255).apply {
-                interpolator = CUSTOM_TILE_ICON_INTERPOLATOR
-                duration = CUSTOM_TILE_ICON_TO_SHOW_MILLIS
-            }.start()
-            iconView.setImageBitmap(screenshot)
+    launch(uiLifecycleCancelJob + UI, CoroutineStart.UNDISPATCHED) {
+        val validUri = item.url.toJavaURI()
+
+        val screenshot = async { HomeTileScreenshotStore.read(itemView.context, item.id) }
+        val title = if (validUri == null) {
+            CompletableDeferred(item.url)
+        } else {
+            async { FormattedDomain.format(itemView.context, validUri, true, 1) }
         }
+
+        // Wait for both to complete so we can animate them together.
+        iconView.setImageBitmap(screenshot.await()) // TODO: if null, provide placeholder.
+        titleView.text = title.await()
+
+        // Animate to avoid pop-in due to thread hand-offs. TODO: animation is janky.
+        AnimatorSet().apply {
+            interpolator = CUSTOM_TILE_ICON_INTERPOLATOR
+            duration = CUSTOM_TILE_TO_SHOW_MILLIS
+
+            val iconAnim = ObjectAnimator.ofInt(iconView, "imageAlpha", 0, 255)
+            val titleAnim = ObjectAnimator.ofFloat(titleView, "alpha", 0f, 1f)
+
+            playTogether(iconAnim, titleAnim)
```

Comment:
Since you're only awaiting on the two results, does it make sense to use an async rather than a launch, because you don't need the Job that's returned?

Reply:
`launch` and `async` are pretty similar: they start a coroutine context and do the work on the specified thread. Where I think they differ is their return value and thus their intent: if you want a return value, use async: e.g. `val screenshot = async.await()`. If you don't and you just want the Job reference to manage the thread, e.g. `job.join()` and `job.cancel()`, use launch.

To answer your question, `launch` doesn't return a result so it'd be tedious to try to get a return value from it.